scout/server/blueprints/variants/views.py

```diff
@@ -184,107 +184,28 @@ def variants(institute_id, case_name):
 
 @variants_bp.route("/<institute_id>/<case_name>/str/variants", methods=["GET", "POST"])
 @templated("variants/str-variants.html")
-def str_variants(institute_id, case_name):
-    """Display a list of STR variants."""
-    page = controllers.get_variants_page(request.form)
-    category = "str"
-    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
-
-    variant_type = Markup.escape(request.args.get("variant_type", "clinical"))
-    if variant_type not in ["clinical", "research"]:
-        variant_type = "clinical"
+def str_variants(institute_id: str, case_name: str):
+    """Display a list of STR variants (STRs)."""
 
-    variants_stats = store.case_variants_count(case_obj["_id"], institute_id, variant_type, False)
-
-    user_obj = store.user(current_user.email)
-
-    if request.method == "POST":
-        if "dismiss_submit" in request.form:  # dismiss a list of variants
-            controllers.dismiss_variant_list(
-                store,
-                institute_obj,
-                case_obj,
-                VARIANT_PAGE,
-                request.form.getlist("dismiss"),
-                request.form.getlist("dismiss_choices"),
-            )
-        form = controllers.populate_filters_form(
-            store, institute_obj, case_obj, user_obj, category, request.form
+    def form_builder(store, inst, case, cat, vtype):
+        """Builds the STRs filters form."""
+        return controllers.populate_sv_mei_str_filters_form(
+            store=store, institute_obj=inst, case_obj=case, category=cat, request_obj=request
         )
-    else:
-        form = StrFiltersForm(request.args)
-
-        if form.gene_panels.data == [] and variant_type == "clinical":
-            form.gene_panels.data = controllers.case_default_panels(case_obj)
-
-        # set form variant data type the first time around
-        form.variant_type.data = variant_type
-        # set chromosome to all chromosomes
-        form.chrom.data = request.args.get("chrom", "")
-
-    controllers.populate_force_show_unaffected_vars(institute_obj, form)
-    controllers.update_form_hgnc_symbols(store, case_obj, form)
-
-    # Populate chromosome select choices
-    controllers.populate_chrom_choices(form, case_obj)
-
-    # Populate custom soft filters
-    controllers.populate_institute_soft_filters(form=form, institute_obj=institute_obj)
-
-    # populate available panel choices
-    form.gene_panels.choices = controllers.gene_panel_choices(store, institute_obj, case_obj)
-
-    controllers.activate_case(store, institute_obj, case_obj, current_user)
-
-    genome_build = get_case_genome_build(case_obj)
-    cytobands = store.cytoband_by_chrom(genome_build)
 
-    query = form.data
-    query["variant_type"] = variant_type
+    def data_exporter(_, case, variants_query):
```

Comment:
store is not needed in download_str_variants, so I include _ for the sake of having a common syntax